### ISSUE TITLE
Handle OSAtomic deprecation

### ISSUE THREAD

**Issue by zwaldowski** [bignerdranch/Deferred#106]
Use os_unfair_lock_t? Remove the thin locks entirely? Remove ReadWriteLock entirely?

**Comment by zwaldowski** [comment]
We'll be able to a variant of the [AtomicSwift]() header in Deferred directly. A crack at this is in the [zwaldowski/integrate-atomics](https://github.com/bignerdranch/Deferred/tree/zwaldowski/integrate-atomics) branch. Once Swift 3.0 Preview 6 lands, I hope to reduce the surface area/impact of these under-the-hood changes even further and clean up the Deferred-side implementation.

**Comment by zwaldowski** [comment]
I still maintain that `LockProtected` is important to the overall usage/friendliness model, but it's becoming not to just recommend `NSLock` or `DispatchSemaphore`. And maybe [`dispatchPrecondition(_:)`](https://github.com/apple/swift/blob/master/stdlib/public/SDK/Dispatch/Dispatch.swift#L40) reduces the need for this whole thing when you can use subsystem-level locking instead.
